src/party.rs

```diff
@@ -24,67 +155,827 @@ pub struct BallotConfig {
+                            })?,
+                            routing: Message2aContent::get_routing(self.id),
+                        })
+                        .map_err(|_| BallotError::Communication("Failed to send Msg2a".into()))?;
+                }
+            }
+            PartyEvent::Launch2av => {
+                if self.status != PartyStatus::Passed2a {
+                    return Err(BallotError::InvalidState(
+                        "Cannot launch 2av, incorrect state".into(),
+                    ));
+                }
+                self.msg_out_sender
+                    .send(MessageWire {
+                        content_bytes: serde_json::to_vec(&Message2avContent {
+                            ballot: self.ballot,
+                            received_value: serde_json::to_vec(&self.value_2a.clone().unwrap())
+                                .map_err(|_| {
+                                    BallotError::MessageParsing(
+                                        "Failed to serialize value for Msg2av".into(),
+                                    )
+                                })?,
+                        })
+                        .map_err(|_| {
+                            BallotError::MessageParsing("Failed to serialize Msg2av".into())
+                        })?,
+                        routing: Message2avContent::get_routing(self.id),
+                    })
+                    .map_err(|_| BallotError::Communication("Failed to send Msg2av".into()))?;
+            }
+            PartyEvent::Launch2b => {
+                if self.status != PartyStatus::Passed2av {
+                    return Err(BallotError::InvalidState(
+                        "Cannot launch 2b, incorrect state".into(),
+                    ));
+                }
+                self.msg_out_sender
+                    .send(MessageWire {
+                        content_bytes: serde_json::to_vec(&Message2bContent {
+                            ballot: self.ballot,
+                        })
+                        .map_err(|_| {
+                            BallotError::MessageParsing("Failed to serialize Msg2b".into())
+                        })?,
+                        routing: Message2bContent::get_routing(self.id),
+                    })
+                    .map_err(|_| BallotError::Communication("Failed to send Msg2b".into()))?;
+            }
+            PartyEvent::Finalize => {
+                if self.status != PartyStatus::Passed2b {
+                    return Err(BallotError::InvalidState(
+                        "Cannot finalize, incorrect state".into(),
+                    ));
+                }
+                self.status = PartyStatus::Finished;
+            }
         }
+        Ok(())
     }
 }
 
-impl<V: Value, VS: ValueSelector<V>> Drop for Party<V, VS> {
-    fn drop(&mut self) {
-        // TODO: stop party.
+#[cfg(test)]
```

Comment:
As tests are starting to consume 50%+ of the file, I would suggest moving message-specific tests to isolated files, one for each message type.

Reply:
I agree that it makes the file a bit bulky, however I would leave these in it, since these tests are intended to check the integrity of the `Party` implementation. Maybe will revisit this stuff upon architectural finalization in M3.